update_versions.yml

```diff
@@ -0,0 +1,116 @@
+---
+
+- name: get webview version
+  hosts: frontend
+  tasks:
+    - shell: cat /var/www/webview/rev.txt
+      register: webview_version
+
+- name: get archive versions
+  hosts: frontend
+  tasks:
+    - shell: /var/cnx/venvs/archive/bin/pip freeze | grep 'cnx-archive'
+      register: archive_version
+      delegate_to: "{{ item }}"
+      with_items:
+        - "{{ groups.archive|first }}"
+    - shell: /var/cnx/venvs/archive/bin/pip freeze
+      register: archive_full_versions
+      delegate_to: "{{ item }}"
+      with_items:
+        - "{{ groups.archive|first }}"
+
+- name: get publishing versions
+  hosts: frontend
+  tasks:
+    - shell: /var/cnx/venvs/publishing/bin/pip freeze | grep 'cnx-publishing'
+      register: publishing_version
+      delegate_to: "{{ item }}"
+      with_items:
+        - "{{ groups.publishing|first }}"
+    - shell: /var/cnx/venvs/publishing/bin/pip freeze
+      register: publishing_full_versions
+      delegate_to: "{{ item }}"
+      with_items:
+        - "{{ groups.publishing|first }}"
+
+- name: get legacy versions
+  hosts: frontend
+  tasks:
+    - shell: echo 'import pip; pip.main(["freeze"])' | /var/lib/cnx/cnx-buildout/bin/instance debug
+      register: legacy_full_versions
+      delegate_to: "{{ item }}"
+      with_items:
+        - "{{ groups.zope|first }}"
+
+- name: get cnx-deploy version
+  hosts: frontend
+  tasks:
+    - shell: git describe --tags
+      register: cnx_deploy_version
+      delegate_to: 127.0.0.1
+
+- name: update version.txt
+  hosts: frontend
+  become: yes
+  become_user: www-data
+  tasks:
+    - shell:
+        cmd: |
+          date +'{"date": "%Y-%m-%d %H:%M:%S %Z",'>/var/www/webview/version.txt
+          cat >>/var/www/webview/version.txt <<EOF
+           "webview": "{{ webview_version.stdout }}",
+           "cnx-archive": "{{ archive_version.results.0.stdout|regex_replace('^[^@]*@([^#]*).*$', '\1')|regex_replace('^[^=]*==') }}",
+           "cnx-publishing": "{{ publishing_version.results.0.stdout|regex_replace('^[^@]*@([^#]*).*$', '\1')|regex_replace('^[^=]*==') }}",
+           "cnx-deploy": "{{ cnx_deploy_version.stdout }}"
+          }
+          EOF
+      args:
+        executable: /bin/sh
+    - name: purge cached version.txt
+      command: "curl -X PURGE 'http://{{ frontend_domain }}:{{ varnish_port }}/version.txt'"
```

Comment:
If I read correctly, the varnish config is currently setup to only accept PURGE requests from localhost. Will this work?

Reply:
Oh... may be not... I'll have to check

Reply:
You're right, it can't be done.  I'll update the PR

Reply:
Any chance we could do that the TODO in the config says, and template in the IPs of the machine where this is running? (we're maybe going to need that for re-baking, eventually anyway)

Reply:
Oh actually the task is executed on the `frontend` hosts and they are the ones with varnish installed. :relieved: so it'll work.

Reply:
@reedstrm I'm not sure what you mean

Reply:
https://github.com/Connexions/cnx-deploy/blob/master/roles/varnish/templates/etc/varnish/varnish.vcl#L110
says "# TODO Include IP addresses of other serving hosts."
This is the ACL. Note that it explicitly defines 127.0.0.1, which is not the network frontend_domain is in. Even though localhost ans 127.0.0.1 are in the acl for PURGE, a request that is not frontend_domain end up catching the hard return of 750. I'll add a oneliner patch.

Reply:
The alternative would be to add branch recognizing incoming from localhost, and allow  that through, then change the PURGE above to `http://localhost`

Reply:
I have `local.cnx.org` pointing to `127.0.0.1`:

```
karen@xenial-openstax:/var/cnx/venvs$ cat /etc/hosts
127.0.0.1       localhost xenial-openstax.local local.cnx.org archive.local.cnx.org authoring.local.cnx.org legacy.local.cnx.org zope.local.cnx.org
```

So it works for me locally:

```
karen@xenial-openstax:/var/cnx/venvs$ curl -X PURGE 'http://local.cnx.org:8990/version.txt'
<!DOCTYPE html>
<html>
  <head>
    <title>200 Ban added</title>
  </head>
  <body>
    <h1>Error 200 Ban added</h1>
    <p>Ban added</p>
    <h3>Guru Meditation:</h3>
    <p>XID: 685955</p>
    <hr>
    <p>Varnish cache server</p>
  </body>
</html>
```

`localhost` doesn't seem to work though:

```
karen@xenial-openstax:/var/cnx/venvs$ curl -X PURGE 'http://localhost:8990/version.txt'
<!DOCTYPE html>
<html>
  <head>
    <title>750 Moved Permanently</title>
  </head>
  <body>
    <h1>Error 750 Moved Permanently</h1>
    <p>Moved Permanently</p>
    <h3>Guru Meditation:</h3>
    <p>XID: 686338</p>
    <hr>
    <p>Varnish cache server</p>
  </body>
</html>
```

Anyway, with your commit, even if `local.cnx.org` doesn't point to `127.0.0.1`, it works so :+1: